src/sentry/profiles/task.py

```diff
@@ -43,6 +48,18 @@ def process_profile(profile: MutableMapping[str, Any], **kwargs: Any) -> None:
         json.dumps(profile),
     )
 
+    track_outcome(
+        org_id=project.organization_id,
+        project_id=project.id,
+        key_id=None,
+        outcome=Outcome.ACCEPTED,
+        reason=None,
+        timestamp=datetime.utcnow().replace(tzinfo=UTC),
+        event_id=profile["transaction_id"],
```

Comment:
We should use the `profile_id` here instead of the `transaction_id`. In practice, I suspect it won't change much since it's just a way to make sure we submit one outcome per event, but since we're talking about profiles here, we should send an ID that is profile related.

Besides that, we might want to move away from transaction in the future and use more the `trace_id`/`span_id` to link a profile to some code so I'd rather not rely on a `transaction_id` if I don't need to.

Thoughts @jan-auer ?

Reply:
also, some context for @jan-auer : right now, the transaction_id we have in the Profile ItemType is the event_id --> `transaction_id==event_id`

Reply:
also, if we end up using `profile_id` instead, I propose to do the same in Relay. We should add `profile_id` in the header as well (of the Profile ItemType) because if in relay we end-up dropping a profile due to bad payload we don't have access to the `profile_id`.

Using `event_id` for the `Outcome::Discard` and the `profiling_id` for the `Outcome::Accepted` doesn't seem ideal.

Reply:
It depends purely on which identifier you would like to search outcomes by. We do support multiple outcomes per event ID and use this e.g. for attachments. Every attachment uses the ID of the event it belongs to, so we can correlate them when searching.

> Besides that, we might want to move away from transaction in the future

In that case, it might be best if you use the profile ID.

> right now, the transaction_id we have in the Profile ItemType is the event_id

That's always the same, transactions are events. In Sentry, there's actually no such thing as `transaction_id`. So this is fine.

>  I propose to do the same in Relay

This is important. At the moment, you're using the transaction's event id (see [here](https://github.com/getsentry/relay/pull/1272/files#diff-4a6c74e3e0ea324e53c16bcb70ef94f5d57730c57dd0401db2e255aedfb119dfR326)), so this would have to change. I'm not sure if you have the profile ID available in all cases in Relay, since it's nowhere in the header. And I'm afraid it may not be practical (or performant) to parse the payload just for that.

Reply:
Alright, let’s keep the `event_id` then.